### ISSUE TITLE
Error: undefined method `undent' for #<String:0x00007fc7ff9f41a0>

### ISSUE THREAD

**Issue by ocyedwin** [d12frosted/homebrew-emacs-plus#110]
Unable to install `emacs-plus` on Mojave, 

```
MBP:Caskroom eong$ brew install --build-from-source emacs-plus
git-lfs clean 'images/screen1.png': git-lfs: command not found
error: external filter 'git-lfs clean %f' failed 127
error: external filter 'git-lfs clean %f' failed
fatal: images/screen1.png: clean filter 'lfs' failed
==> Installing emacs-plus from d12frosted/emacs-plus
==> Downloading https://ftp.gnu.org/gnu/emacs/emacs-25.3.tar.xz
Already downloaded: /Users/eong/Library/Caches/Homebrew/downloads/a5e71c64f65cbb475ebab05f02d1743bab48b4d0c14295c8bd9532534c545b23--emacs-25.3.tar.xz
==> Downloading https://gist.githubusercontent.com/aatxe/260261daf70865fbf1749095de9172c5/raw/214b50c62450be1cbee9
Already downloaded: /Users/eong/Library/Caches/Homebrew/downloads/7dccd5b8ea0bb2eb29cfb643365489d84c16ea33b1743261be66fa9691733a1e--patch-multicolor-font.diff
==> Downloading https://gist.githubusercontent.com/aaronjensen/f45894ddf431ecbff78b1bcf533d3e6b/raw/6a5cd7f57341ab
Already downloaded: /Users/eong/Library/Caches/Homebrew/downloads/80498ff2d7eb575fe818184e1576c40a903f3d15aad940a43772ea3754924a45--Emacs-25-OS-X-use-vfork.patch
==> Patching
==> Applying patch-multicolor-font.diff
patching file src/macfont.m
==> Applying Emacs-25-OS-X-use-vfork.patch
patching file src/callproc.c
patching file src/conf_post.h
patching file src/process.c
==> ./configure --disable-silent-rules --enable-locallisppath=/usr/local/share/emacs/site-lisp --infodir=/usr/loca
==> make
==> make install
######################################################################## 100.0%
Error: An exception occurred within a child process:
  NoMethodError: undefined method `undent' for #<String:0x00007fd59a470e38>
```

debug info:
```
Ongs-MBP:Caskroom eong$ brew --config
HOMEBREW_VERSION: 1.8.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 38493a35a2abe722c527ffebf9c2fbf9bc4d3825
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e963aa704fe743f54d34bff943aaed7a0175f668
Core tap last commit: 7 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_AWS_ACCESS_KEY_ID: set
HOMEBREW_AWS_SECRET_ACCESS_KEY: set
HOMEBREW_VISUAL: emacsclient -c -a emacs
CPU: octa-core 64-bit skylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.20.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 1.8.0_152
macOS: 10.14-x86_64
CLT: 10.1.0.0.1.1539992718
Xcode: 10.1
```

`brew info` gave same error:

```
MBP:Caskroom eong$ brew info emacs-plus
d12frosted/emacs-plus/emacs-plus: stable 25.3 (bottled), devel 26.0.90, HEAD
GNU Emacs text editor
https://www.gnu.org/software/emacs/
Not installed
From: https://github.com/d12frosted/homebrew-emacs-plus/blob/master/Formula/emacs-plus.rb
==> Dependencies
Build: pkg-config ✔
Recommended: gnutls ✔, librsvg ✔, imagemagick@6 ✔
Optional: dbus ✘, mailutils ✘
==> Requirements
Optional: x11 ✘
==> Options
--with-24bit-color
	Experimental: build with 24 bit color support (stable only)
--with-ctags
	Don't remove the ctags executable that Emacs provides
--with-dbus
	Build with dbus support
--with-mailutils
	Build with mailutils support
--with-natural-title-bar
	Experimental: use a title bar colour inferred by your theme (stable only)
--with-no-title-bars
	Experimental: build with a patch for no title bars on frames (--HEAD and --devel has this built-in via undecorated flag)
--with-pixel-scrolling
	Build with a patch from emacs-mac supporting native pixel scrolling (stable only)
--with-x11
	Experimental: build with x11 support
--without-cocoa
	Build a non-Cocoa version of Emacs
--without-gnutls
	Build without gnutls support
--without-imagemagick@6
	Build without imagemagick@6 support
--without-librsvg
	Build without librsvg support
--without-libxml2
	Build without libxml2 support
--without-modules
	Build without dynamic modules support
--without-multicolor-fonts
	Build without a patch that enables multicolor font support
--without-spacemacs-icon
	Build without Spacemacs icon by Nasser Alshammari
--devel
	Install development version 26.0.90
--HEAD
	Install HEAD version
Error: undefined method `undent' for #<String:0x00007fc7ff9f41a0>
Please report this bug:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Taps/d12frosted/homebrew-emacs-plus/Formula/emacs-plus.rb:257:in `plist'
/usr/local/Homebrew/Library/Homebrew/extend/os/mac/caveats.rb:6:in `plist_caveats'
/usr/local/Homebrew/Library/Homebrew/caveats.rb:26:in `caveats'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/forwardable.rb:202:in `empty?'
/usr/local/Homebrew/Library/Homebrew/cmd/info.rb:208:in `info_formula'
/usr/local/Homebrew/Library/Homebrew/cmd/info.rb:80:in `block in print_info'
/usr/local/Homebrew/Library/Homebrew/cmd/info.rb:69:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/info.rb:69:in `each_with_index'
/usr/local/Homebrew/Library/Homebrew/cmd/info.rb:69:in `print_info'
/usr/local/Homebrew/Library/Homebrew/cmd/info.rb:56:in `info'
/usr/local/Homebrew/Library/Homebrew/brew.rb:89:in `<main>'
```

**Comment by d12frosted** [comment]
Hi,

Sorry for the long silence.

Are you sure you didn't perform any changes to the bottle or to the sources? AFAIK, Emacs doesn't use `git-lfs`. Nor do I use it in this repo.

As a workaround, try to install `git-lfs` by doing `brew install git-lfs`.

P. S. I wonder why `emacs-plus` installs Emacs 25. Please run `brew update` before installing anything.

**Comment by d12frosted** [comment]
Closing for lack of activity. Let me know if you still have the problem.

**Comment by braham-snyder** [comment]
For those who run into an error like this: I just fixed a similar error (different `String`, also said `Error: Failed to install plist file`, and different but still outdated `emacs-plus` formula being used). Uninstalling and reinstalling didn't help (still got old formula), nor did `brew cleanup`. But `brew untap d12frosted/emacs-plus` and `brew cleanup -s` followed by `brew install emacs-plus` got the new formula and fixed the issue. My best guess as to what happened is something roughly like: I had edited the old formula using `brew edit` (fairly certain I had done this a long time ago), and so my edited formula was overriding the new formula.
